src/QuickSpec/Internal/Haskell.hs

```diff
@@ -256,8 +256,10 @@ instance Observe t p a => Observe t (Maybe p) (DM.First a) where
   observe t = observe t . DM.getFirst
 instance Observe t p a => Observe t (Maybe p) (DM.Last a) where
   observe t = observe t . DM.getLast
+#if !MIN_VERSION_base(4,13,0)
```

Comment:
A small point:

It looks like  `Option` still existed in that version of `base`. It seems to have been removed in [version 4.16.0](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Semigroup.html).  In the immediately preceding version 4.15.1.0, [it still existed](https://hackage.haskell.org/package/base-4.15.1.0/docs/Data-Semigroup.html#t:Option)